jni/test/jni_array_test.dart

```diff
@@ -0,0 +1,212 @@
+import 'dart:io';
+
+import 'package:jni/jni.dart';
+import 'package:test/test.dart';
+
+void main() {
+  // Don't forget to initialize JNI.
+  if (!Platform.isAndroid) {
+    try {
+      Jni.spawn(dylibDir: "build/jni_libs", jvmOptions: ["-Xmx128m"]);
+    } on JvmExistsException catch (_) {
+      // TODO(#51): Support destroying and reinstantiating JVM.
+    }
+  }
+  test("Java boolean array", () {
+    final array = Jni.newBooleanArray(3);
+    expect(array.length, 3);
+    array[0] = true;
+    array[1] = false;
+    array[2] = false;
+    expect(array[0], true);
+    expect(array[1], false);
+    expect(array[2], false);
+    array.setRange(0, 3, [false, true, true, true], 1);
+    expect(array[0], true);
+    expect(array[1], true);
+    expect(array[2], true);
+    expect(() {
+      array[-1] = false;
+    }, throwsRangeError);
+    expect(() {
+      array[3] = false;
```

Comment:
`_ = array[3]` if you want to ensure both getter and setter are checking range errors.

Reply:
Will add this 👍